services/bigid-azure-key-vault-integration-app/src/main/java/com/bigid/azurekeyvaultapp/service/impl/FetchCredentialsExecutionService.java

```diff
@@ -0,0 +1,83 @@
+import com.bigid.appinfrastructure.dto.StatusEnum;
+import com.bigid.appinfrastructure.externalconnections.BigIDProxy;
+import com.bigid.appinfrastructure.services.AbstractExecutionService;
+import com.bigid.azurekeyvaultapp.constant.ActionParams;
+import com.bigid.azurekeyvaultapp.service.ExecutionService;
+import com.bigid.azurekeyvaultapp.service.KeyVaultTokenService;
+import com.bigid.azurekeyvaultapp.util.ParamsMapUtils;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.tuple.Triple;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Mono;
+
+import java.util.Map;
+
+@Service
+@Slf4j
+public class FetchCredentialsExecutionService extends AbstractExecutionService implements ExecutionService {
+
+    private static final String SECRET_KEY = "secret_key";
+    private static final String SUCCESSFULLY_FETCHED_SECRET = "Successfully fetched secret";
+    public static final String FAILED_TO_FETCH_SECRET = "Failed to fetch secret";
+
+    private final KeyVaultTokenService keyVaultTokenService;
+
+    @Autowired
+    public FetchCredentialsExecutionService(BigIDProxy bigIDProxy, KeyVaultTokenService keyVaultTokenService) {
+        super(bigIDProxy);
+        this.keyVaultTokenService = keyVaultTokenService;
+    }
+
+    @Override
+    public Triple<Boolean, String, Map<String, String>> performAction(ExecutionContext executionContext) {
+        try {
+            AccessToken accessToken = keyVaultTokenService.fetchAccessToken(executionContext);
+            TokenCredential tokenCredential = request -> Mono.just(accessToken);
+
+            Map<String, Object> actionParamsMap = ParamsMapUtils.getActionParamsMap(executionContext);
+            SecretClient secretClient = new SecretClientBuilder()
+                    .vaultUrl(actionParamsMap.get(ActionParams.AZURE_KEY_VAULT_URL.getValue()).toString())
+                    .credential(tokenCredential)
+                    .buildClient();
+
+            // Fetch the secret from Azure Key Vault
+            ObjectMapper objectMapper = new ObjectMapper();
+            JsonNode rootNode = objectMapper.readTree(actionParamsMap.get(ActionParams.CREDENTIAL_PROVIDER_CUSTOM_QUERY.getValue()).toString());
+            String secretKey = rootNode.get(SECRET_KEY).asText();
+
+            String secretValue = secretClient.getSecret(secretKey).getValue();
+            bigIDProxy.updateActionStatusToBigID(
+                    executionContext,
+                    initializeResponse(executionContext, StatusEnum.COMPLETED, 1d, SUCCESSFULLY_FETCHED_SECRET)
+            );
+            log.info(SUCCESSFULLY_FETCHED_SECRET);
+
+            return Triple.of(true, SUCCESSFULLY_FETCHED_SECRET, Map.of(secretKey, secretValue));
```

Comment:
value returned should have shape: { credentialFields: {secretValue} } with credentialFields key word (assuming secretValue is the key:value set of the returned secret